### PR TITLE
CI: Introduce docs pipeline

### LINKED ISSUE #40703: CI/Build: Create standalone docs pipeline
**What happened**:

Currently, for every docs change, the whole pipeline runs, including building-testing frontend and backend.
What we'd really need ideally, would be a standalone pipeline which should run only for docs changes.

### PR DESCRIPTION
**Pull request by dsotirakis**
**What this PR does / why we need it**:

We are introducing a standalone docs pipeline which will run only for docs changes on every PR.

Only `pr-docs` pipeline will run upon documentation changes. In case there are both docs and code changes, the regular 3 pipelines will run (build, tests, integration tests) adding `pr-docs` to them (4 in total)

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/40703

**Special notes for your reviewer**:

This PR needs to be merged for changes to be reflected.

### CODE REVIEW

File: scripts/drone/pipelines/pr.star [hunk 1]
```diff
@@ -116,7 +112,15 @@ def pr_pipelines(edition):
...
 
     trigger = {
-        'event': ['pull_request',],
+        'event': [
+            'pull_request',
+        ],
+        'paths': {
+            'exclude': [
+                'docs/**',
+                'packages/**',
```
Comment (mckn) [inline scripts/drone/pipelines/pr.star:121]: I think the `packages/**` should be removed here. We should run the test-pr pipeline if we have changes in the `packages/**` folder.

**Review by mckn — CHANGES_REQUESTED** [review]
I'm might read this incorrect but I think the `packages/**` should not be excluded from the `test-pr` pipeline.

**Comment by dsotirakis** [comment]
Hm okay so let me write down a use case:

1. Changes to a doc file (under `docs/`):
* **ONLY** `pr-docs` pipeline should run.
2. Changes to a packages file (under `packages/`):
* **BOTH** `pr-docs` and the rest of the pipelines that already run on every pr should run.
3. Changes to both files
* **BOTH** `pr-docs` and the rest of the pipelines that already run on every pr should run.

Am I right?

**Comment by mckn** [comment]
> Hm okay so let me write down a use case:
> 
> 1. Changes to a doc file (under `docs/`):
> 
> * **ONLY** `pr-docs` pipeline should run.
> 
> 2. Changes to a packages file (under `packages/`):
> 
> * **BOTH** `pr-docs` and the rest of the pipelines that already run on every pr should run.
> 
> 3. Changes to both files
> 
> * **BOTH** `pr-docs` and the rest of the pipelines that already run on every pr should run.
> 
> Am I right @mckn?

Yes - that sounds correct.

**Review by mckn — APPROVED** [review]
LGTM!

**Pull request merged**